hudi-client/pom.xml

```diff
@@ -68,6 +107,12 @@
   </build>
 
   <dependencies>
+    <!-- Scala -->
+    <dependency>
+      <groupId>org.scala-lang</groupId>
```

Comment:
should we limit scala to just the spark module?

Reply:
> should we limit scala to just the spark module?
Yes, it is better. can we do it in another PR?
because although some classes have nothing to do with spark, it used `scala.Tuple2`, so scala is still needed. 
we can replace it with 'org.apache.hudi.common.util.collection.Pair'
WDYT?

Reply:
fair. let me take a closer look